rosbag2/src/rosbag2/writer.cpp

```diff
@@ -76,6 +89,13 @@ void Writer::create_topic(const TopicMetadata & topic_with_type)
     converter_->add_topic(topic_with_type.name, topic_with_type.type);
   }
 
+  if (topics_.find(topic_with_type.name) == topics_.end()) {
+    rosbag2_storage::TopicInformation info{};
+    info.topic_metadata = topic_with_type;
+
+    topics_.insert({topic_with_type.name, info});
+  }
+
   storage_->create_topic(topic_with_type);
```

Comment:
with the metadata handled in here, does it make sense to still call `storage_->create_topic(..)`?

Reply:
Yeah, in the case of sqlite, it writes the topic name and serialization format into a table

Reply:
Sorry my bad. I wasn't precise enough. I was thinking to call `create_topic` only once per topic, essentially move this call up into the `if` block